### PR TITLE
ISPN-5476 Cross-site tests should run in parallel

### PR DESCRIPTION
**Pull request by danberindei**
https://issues.jboss.org/browse/ISPN-5476

May also fix https://issues.jboss.org/browse/ISPN-7024

Create a TEST_RELAY2 protocol that allows setting a different bridge cluster name for each test.

### CODE REVIEW

**Comment by tristantarrant** [comment]
This doesn't fix ISPN-7024 since SiteDownFailoverTest inherits from AbstractXSiteTest and you've only patched AbstractTwoSitesTest.

**Comment by tristantarrant** [comment]
Ignore my completely useless comment above, as the changes to AbstractXSiteTest have nothing to do with the overall fix. However, I have seen  SiteDownFailoverTest fail locally with your PR

**Comment by tristantarrant** [comment]
Apologies, I had not installed infinispan-core. I should stop commenting out of my ...

**Comment by tristantarrant** [comment]
Actually, this seems to fix the issue.

**Comment by pruivo** [comment]
should the test-xsite profile being removed? I guess it was there only to make sure the test runs sequentially...

File: core/src/test/java/org/infinispan/test/fwk/TEST_RELAY2.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      super.parseSiteConfiguration(map);
+
+      String testName = TestResourceTracker.getCurrentTestName();
+      map.forEach((s, siteConfig) -> {
+         List<RelayConfig.BridgeConfig> bridges = siteConfig.getBridges();
+         for (int i = 0; i < bridges.size(); i++) {
+            RelayConfig.BridgeConfig bridgeConfig = bridges.get(i);
+            String config =
+                  (String) TestingUtil.extractField(RelayConfig.PropertiesBridgeConfig.class, bridgeConfig, "config");
+            if (testName != null) {
```
Comment (pruivo) [inline core/src/test/java/org/infinispan/test/fwk/TEST_RELAY2.java]: I guess you can check the `testName` before the `forEach()`

**Comment by pruivo** [comment]
`NonTxOfflineTest.testPutWithFailures` is failing when I use more than 1 thread

**Comment by danberindei** [comment]
@pruivo I haven't seen `NonTxOfflineTest` fail locally, but I'm making the keys an instance field just in case.
I've was hesitating to remove the `test-xsite` profile, but I guess there's no point in delaying it.

File: core/src/test/java/org/infinispan/test/fwk/TEST_RELAY2.java (hunk 1 above)
Comment (danberindei) [inline core/src/test/java/org/infinispan/test/fwk/TEST_RELAY2.java]: I updated the code to change the cluster name in both cases.

**Comment by pruivo** [comment]
`NonTxOfflineTest` seems fine now...
You have a new failure in cli-interpreter `SiteStatementTest`. I guess it was never run before... 
* The test `testContainerOperations()` needs to define `"another-cache"` in NYC site, otherwise it will ignore the xsite command (it returns a `CacheNotFoundResponse`)
* I'm unable to reproduce `testSiteStateTransfer()` locally... so no idea what is going on...

**Comment by danberindei** [comment]
Updated to fix `SiteStatementTest`. I haven't reproduced `testSiteStateTransfer()` locally either, let's wait for a couple more runs in CI.

**Comment by danberindei** [comment]
Updated again to add a method `AbstractXSiteTest.waitForSites()` and improve logging for failed xsite state push.

File: core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java [hunk 2]
```diff
@@ -280,7 +280,7 @@ public final String pushState(@Parameter(description = "The destination site nam
       try {
          stateTransferManager.startPushState(siteName);
       } catch (Throwable throwable) {
-         log.debugf(throwable, "Unable to pushState to '%s'.", siteName);
+         log.errorf(throwable, "Unable to pushState to '%s'.", siteName);
```
Comment (pruivo) [inline core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java]: it should be in `Log` interface.

Comment (danberindei) [inline core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java]: @pruivo I'd rather keep it the same as the error string that's returned to the user.

Comment (pruivo) [inline core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java]: don't we have a policy that all error messages should be i8n?

Comment (danberindei) [inline core/src/main/java/org/infinispan/xsite/XSiteAdminOperations.java]: True... ok, I've i18n-ed the error message and added it to the other operations as well.

**Comment by pruivo** [comment]
:+1:

**Comment by tristantarrant** [comment]
Merged, thanks

**Pull request merged**